proposals/nnn-audit-logging.md

```diff
@@ -0,0 +1,190 @@
+
+#### Proxy-scoped events
+
+* `ProxyStartup` — Emitted when the proxy starts up, before it binds to any sockets.
+    - `processUuid` — identifies this process uniquely in time and space.
+    - `instanceName` — is optionally provided by the user to identify this instance.
+    - `currentTimeMillis` — The number of milliseconds since the UNIX epoch.
+    - `hostName` — The name of the host on which the proxy is running.
+* `ProxyCleanShutdown` — Emitted when the proxy shuts down normally. Obviously it's not possible to emit anything in the case of a crash (e.g. `SIGKILL`).
+  The absence of a `ProxyCleanShutdown` in a stream of events with the same `processUuid`
+  would indicate a crash (or that the process is still alive). 
+    - `processAgeMicros` — The time of the event, measured as the number of microseconds since proxy startup.
+
+#### Session-scoped events
+
+Session-scoped events all have at least the following  attributes: 
+- `processAgeMicros` — The time of the event, measured as the number of microseconds since proxy startup.
+- `sessionId` — A UUID that uniquely identifies the session in time and space.
+
+* `ClientAccept` — Emitted when the proxy accepts a connection from a client
+    - `processUuid` — Allows sessions of the same proxy process to be correlated.
+    - `virtualCluster` — The name of the virtual cluster the client is connecting to
+    - `peerAddress` — The IP address and port number of the remote peer.
+* `BrokerConnect` — Emitted when the proxy connects to a broker
+    - `brokerAddress` — The IP address and port number of the remote peer.
+* `ClientSaslAuthFailure` — Emitted when a client completes SASL authentication unsuccessfully
+    - `attemptedAuthorizedId` — The authorized id the client attempted to use, if known.
+* `ClientSaslAuthSuccess` — Emitted when a client completes SASL authentication successfully
+    - `authorizedId` — The authorised id
+* `OperationAllowed` — Emitted when an `Authorizer` allows access to a resource.
+    - `op` — The operation that was allowed (e.g. `READ`)
+    - `resourceType` — The type of the resource (e.g. `Topic`)
+    - `resourceName` — The name of the resource (e.g. `my-topic`
+* `OperationDenied` — Emitted when an `Authorizer` denies access to a resource.
+    - `op` — The operation that was denied (e.g. `READ`)
+    - `resourceType` — The type of the resource (e.g. `Topic`)
+    - `resourceName` — The name of the resource (e.g. `my-topic`
+* `Read` — Emitted when a client successfully reads records from a topic. It is called `Read` rather than `Fetch` because it covers reads generally, including the `ShareFetch` API key. It will be possible to disable these events, because of the potential for high volume. 
+    - `topicName` — The name of the topic.
```

Comment:
what about the audit of client ids, group names and possibly, transactional ids?